bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/WlanThermoMiniHandler.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wlanthermo.internal;
+
+import java.net.URISyntaxException;
+import java.util.concurrent.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.thing.*;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.wlanthermo.internal.api.mini.builtin.App;
+import org.openhab.binding.wlanthermo.internal.api.mini.builtin.WlanThermoMiniCommandHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link WlanThermoMiniHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Schlipp - Initial contribution
+ */
+@NonNullByDefault
+public class WlanThermoMiniHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(WlanThermoMiniHandler.class);
+    private final WlanThermoMiniCommandHandler wlanThermoMiniCommandHandler = new WlanThermoMiniCommandHandler();
+
+    private WlanThermoMiniConfiguration config = new WlanThermoMiniConfiguration();
+    private final HttpClient httpClient;
+    private @Nullable ScheduledFuture<?> pollingScheduler;
+    private final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(WlanThermoBindingConstants.WLANTHERMO_THREAD_POOL);
+    private final Gson gson = new Gson();
+    private App app = new App();
+
+    public WlanThermoMiniHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing WlanThermo Mini!");
+        config = getConfigAs(WlanThermoMiniConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.schedule(this::checkConnection, config.getPollingInterval(), TimeUnit.SECONDS);
+
+        logger.debug("Finished initializing WlanThermo Mini!");
+    }
+
+    private void checkConnection() {
+        try {
+            if (httpClient.GET(config.getUri("/app.php")).getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+                if (pollingScheduler != null) {
+                    pollingScheduler.cancel(false);
+                }
```

Comment:
```suggestion
                ScheduledFuture<?> oldScheduler = pollingScheduler;
                if (oldScheduler != null) {
                    oldScheduler.cancel(false);
                }
```

This should fix

```
[WARNING] /home/travis/build/openhab/openhab-addons/bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/WlanThermoMiniHandler.java:[76,2910] Potential null pointer access: this expression has a '@Nullable' type
```

Also applies below in l. 88-90, l. 129-131 and l.141.143. In case you do not immediately re-init the scheduler, set the `pollingScheduler = null;`. Probably also in the other handler.

Reply:
That indeed removed the warning. Changed it!